src/odemis/driver/npmc.py

```diff
@@ -140,69 +111,73 @@ def __init__(self, name, role, port, unit='s', axes=None, offset=None, inverted=
                 axis_range = (-1, 1)
 
             try:
-                axis_speed_range = axis_par['speed']
+                axis_unit = axis_par['unit']
             except KeyError:
-                logging.info("Axis %s has no speed range. Assuming (10e-6, 0.5)", axis_name)
-                axis_speed_range = (10e-6, 0.5)
-
-            # deal with unit conversion
-            if self._axis_unit == 'm':
-                self.SetAxisUnit(axis_num, 'mm')
-                self._axis_conv_factor[axis_num] = 1000
-            elif self._axis_unit == 's':  # amount of delay in the delay stage.
-                self.SetAxisUnit(axis_num, 'mm')
-                # path length is double since it is reflected by a mirror
-                self._axis_conv_factor[axis_num] = C / 2
-            elif self._axis_unit == 'mm':
-                self.SetAxisUnit(axis_num, self._axis_unit)
-                self._axis_conv_factor[axis_num] = 1
-            else:
-                raise ValueError("Axis %s has an invalid unit %s." % (axis_name, self._axis_unit))
-                
+                logging.info("Axis %s has no unit. Assuming m", axis_name)
+                axis_unit = "m"
+
```

Comment:
Please, for each axis, use `ID?` and put it in `._hwVersion` (as a concatenation). Also, put VE? into `_swVersion`.